packages/loader/container-loader/src/audience.ts

```diff
@@ -30,8 +30,9 @@ export class Audience extends EventEmitter implements IAudience {
      * Removes a client from the audience
      */
     public removeMember(clientId: string) {
+        const removed = this.members.get(clientId);
```

Comment:
So as is one weird thing that can happen is anyone can remove any clientId from the audience.  I reflected that in the IClient sent into the event here, but do we think it's worth guarding against this by only emitting when something was actually removed?

Reply:
But we removing here in this function by calling delete on the members. Looks like you are just adding client to the event and not changing any functionality.

Reply:
Yeah there's no functionality change here as is.  But anyone with access to the audience could potentially call `removeMember` and the audience will emit a `removeMember` event even for a bogus `clientId` which seems weird.

Reply:
IAudience (exposed to consumers) does not have methods to manipulate audience, only observe changes.

Reply:
Oh, that's true.  I think it's still possible (but very unlikely) for this to happen though because audience additions/removals are signal based (if you miss the addition signal but get the removal signal due to lack of signal guarantees).  @vladsud Do we typically want to handle these sorts of issues internally or transmit them out as usual for users to handle?

Reply:
If I got you right, you are suggesting that signals maybe lost within single (websocket) connection session, and thus we may not get addition, but only get removal signal.
I believe it's true, but I'm not sure what's the right handling here :) I.e. I think our code is not hardened against it, and we should likely harden it (i.e. not raise event for "remove" if we never had that client in audience). But I'm not sure it's enough, i.e. I'm not sure how this inconsistency reflects to code that uses audience. Will it backfire?

I'm pretty sure (based on observing race conditions and complicities on server side around ordered delivery of ops), that it would be almost impossible to provide stronger guarantees here for signals, so we should bite the bullet and assume that indeed things can go out of order and we need to live with it. Adding telemetry when such things happen would be nice.
If we see it being hit with some frequency, I think we would need to drill into it and understand better.

Reply:
I went ahead and changed it not to emit when there was nothing removed, and to add a telemetry error when that happened (though not entirely sure if error is right here given this behavior is not actually outside expectations for signals).  

As for users, I think they will largely be unaffected in this case.  Given there is a missing audience member, users might be accumulating data about a client (and presumably also from that client) that doesn't exist in the audience.  There might be some inconsistencies depending on how this gets handled/displayed, which we should narrow through consistent examples/guidance on usage.  Losing the `removeMember` event would mean users would retain information on that disconnecting client, which leads to essentially the same situation as before that client disconnected.

All that said, it sounds like quorum is doing the same thing as audience except by using ops.  I'm assuming all the same clients get added to both (not sure on this), in which case why keep two things except one is less reliable than the other?

Reply:
quorum contains only "write" clients, while audience contains all clients. "read" clients can be clients who have only read permissions, as well as clients who did not submit any op (and thus did not have a need to convert to "write" connection).